### PR TITLE
Show meaningful error when updating a single element of array

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -27,6 +27,9 @@ Changes
 Fixes
 =====
 
+ - Improved error message when trying to update an element of an array using 
+   the index in the array.
```
Comment (seut) [inline CHANGES.txt]: would skip the ".. using.." part, afaik there is no other way updating an array element

File: sql/src/main/java/io/crate/analyze/UpdateAnalyzer.java [hunk 2]
```diff
@@ -193,4 +205,20 @@ private boolean hasMatchingParent(TableInfo tableInfo, Reference info, Predicate
...
+        @Override
+        protected Void visitSubscriptExpression(SubscriptExpression node, Boolean context) {
+            process(node.index(), true);
+            return super.visitSubscriptExpression(node, context);
+        }
+
+        @Override
+        protected Void visitLongLiteral(LongLiteral node, Boolean context) {
+            if (context) {
+                throw new IllegalArgumentException("Updating a single element of array is not supported");
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/UpdateAnalyzer.java]: of **an** array

File: sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java [hunk 3]
```diff
@@ -381,17 +382,32 @@ public void testUpdateNestedObjectDeleteWithoutArgs() throws Exception {
...
         refresh();
 
         expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("Array index must be in range 1 to 2147483648");
+        expectedException.expectMessage("Updating a single element of array is not supported");
+        execute("update test set coolness[1] = 3.3");
+    }
+
+    @Test
+    public void testUpdateArrayUsingArrayCatFunction() {
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: hm as col = array_cat() is just processed like any expression assignment, I'm not sure if we need a dedicated test for it

File: sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java [hunk 4]
```diff
@@ -381,17 +382,32 @@ public void testUpdateNestedObjectDeleteWithoutArgs() throws Exception {
     }
 
     @Test
-    public void testUpdateWithNestedObjectArrayIdxAccess() throws Exception {
+    public void testUpdateArrayElementIsNotAllowed() {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: Can we write this as an analyzer unit test?

File: sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java [hunk 5]
```diff
@@ -381,17 +382,32 @@ public void testUpdateNestedObjectDeleteWithoutArgs() throws Exception {
     }
 
     @Test
-    public void testUpdateWithNestedObjectArrayIdxAccess() throws Exception {
```
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: Don't remove this test (with the range), maybe move it to SelectStatementAnalyzerTest.
@seut what do you think?

File: sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java (hunk 4 above)
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: And I'd move this one as a "Unit test" to UpdateAnalyzerTest. @seut what do you think?

File: sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java (hunk 5 above)
Comment (andreidan) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: There is SelectStatementAnalyzerTest.testSubscriptArrayInvalidIndexMin

Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java]: why not removing it? is not supported despite of range (integer vs long).

**Pull request merged**